### PR TITLE
settings/core: Improve ssl settings

### PR DESCRIPTION
**Pull request by sirainen**

### CODE REVIEW

File: source/configuration_manual/dovecot_ssl_configuration.rst [hunk 1]
```diff
@@ -401,9 +401,9 @@ Testing CA Success
 Client connections
 ******************
 
-.. versionadded:: v2.3.4 
-
 Dovecot uses default system CAs for outgoing connections.
+The :ref:`setting-ssl_client_ca_dir` is preferred because it uses less memory.
```
Comment (cmouse) [inline source/configuration_manual/dovecot_ssl_configuration.rst:405]: Could indicate that using System CAs is default behavior.

File: source/settings/core.rst [hunk 2]
```diff
@@ -4159,15 +4165,24 @@ to apply a security update, for example.
 
 The level of SSL support.
 
-ssl=no: SSL/TLS is completely disabled.
-
-With both ssl=yes and ssl=required it's still possible that the client attempts to do a plaintext authentication before enabling SSL/TLS, which exposes the plaintext password to the internet. Dovecot attempts to indicate this to the IMAP clients via the LOGINDISABLED capability, but many clients still ignore it and send the password anyway. There is unfortunately no way for Dovecot to prevent this behavior. The POP3 standard doesn't have an equivalent capability at all, so the POP3 clients can't even know if the server would accept a plaintext authenticatio
+ * ``no``: SSL/TLS is completely disabled.
+ * ``yes``: SSL/TLS is enabled.
+ * ``required``: SSL/TLS is required for all imap, pop3, managesieve and
```
Comment (cmouse) [inline source/settings/core.rst:4170]: Does this allow STARTTLS still?

**Review by cmouse — CHANGES_REQUESTED** [review]
We should fix our terminology to use TLS instead of SSL in text.

**Comment by sirainen** [comment]
>We should fix our terminology to use TLS instead of SSL in text.

This is a bit troublesome. All the configuration is using ssl_ prefixes. Example config also talks about SSL. Also "SSL certificate" has 37M hits in google, while "TLS certificate" has only 500k hits.

File: source/configuration_manual/dovecot_ssl_configuration.rst (hunk 1 above)
Comment (sirainen) [reply source/configuration_manual/dovecot_ssl_configuration.rst]: I changed the text here. I'm not actually sure how exactly the openssl system CAs work. Are they built-in or read from some default directory or what..

File: source/settings/core.rst (hunk 2 above)
Comment (sirainen) [reply source/settings/core.rst:4171]: Added a sentence about this.

**Review by cmouse — APPROVED** [review]

**Pull request merged**